src/components/GoogleLogin/index.tsx

```diff
@@ -1,20 +1,80 @@
-import React, { ReactElement, useCallback, useMemo } from 'react';
+import {
+  GoogleSignin,
+  statusCodes,
+} from '@react-native-community/google-signin';
+import React, { ReactElement, useMemo } from 'react';
+import { getUniqueId } from 'react-native-device-info';
 
 import SocialIcon from '@src/components/SocialIcon';
+import { flash } from '@src/functions';
+import { useSocialLoginMutation } from '@src/operations/mutations/socialLogin';
 import { SocialProvider } from '@src/types/graphql';
 
 import { SNSButton } from './styled';
 
 type P = {};
 
+GoogleSignin.configure({
+  webClientId: process.env.GOOGLE_WEB_CLIENT_ID,
+  offlineAccess: false,
+  forceCodeForRefreshToken: false,
+  iosClientId: process.env.GOOGLE_IOS_CLIENT_ID,
+});
+
 const GoogleLogin: React.FC<P> = () => {
   const node = useMemo<ReactElement>(
     () => <SocialIcon provider={SocialProvider.Google} />,
     [],
   );
-  const onGoogleLogin = useCallback(() => {}, []);
 
-  return <SNSButton node={node} onPress={onGoogleLogin} color="grey5" />;
+  const [socialLogin] = useSocialLoginMutation();
+
+  const signIn = async () => {
+    try {
+      await GoogleSignin.hasPlayServices();
+      const { user } = await GoogleSignin.signIn();
+
+      await socialLogin({
+        variables: {
+          input: {
+            deviceID: getUniqueId(),
+            email: user.email,
+            name: user.name || '',
```

Comment:
`user.name`이 `undefined` 일 수 없다면 '' 빼도 될거 같아요~

```suggestion
            name: user.name,
```

Reply:
구글 로그인에서 return 해주는 user 데이터의 name 이 nullable 이네요 ! 
이럴 경우 위처럼 말고 다른 좋은 방법이 있을까요 ?

<img width="334" alt="image" src="https://user-images.githubusercontent.com/43168524/184899837-746125fa-9b33-44e8-a148-3e22b8788091.png">

Reply:
아〰️〰️ 그렇군요! 음 `name`을 if문으로 확인한다거나 아님 `user.name as string` 등으로 선언할 수 도 있겠네요!
지금 지걸님의 방법으로도 좋을것 같네요!